syntax = "proto3";

service ServiceManager {
    rpc restart_service (ServiceRequest) returns (ServiceResponse);
    rpc start_service (ServiceRequest) returns (ServiceResponse);
    rpc stop_service (ServiceRequest) returns (ServiceResponse);
    rpc search_service (SearchRequest) returns (SearchResponse);
    rpc status_service (ServiceRequest) returns (StatusResponse);
    rpc logs_service (LogsRequest) returns (LogsResponse);
}

message ServiceRequest {
    string service_name = 1;
}

message ServiceResponse {
    string status = 1;
}

message SearchRequest {
    string search_term = 1;
}

message SearchResponse {
    repeated string container_names = 1;
}

message StatusResponse {
    repeated string statuses = 1;
}

message LogsRequest {
    string service_name = 1;
    bool follow = 2;
}

message LogsResponse {
    repeated string logs = 1;
}